netsim/modules/lag.py

```diff
@@ -0,0 +1,75 @@
+import typing
+import netaddr
+
+from box import Box, BoxList
+from . import _Module
+from .. import data
+from ..utils import log
+from ..augment import devices
+
+#
+# Checks if 2 lists have the same elements, independent of order
+#
+def same_list(l1:BoxList,l2:BoxList) -> bool:
+  for l in l1:
+    if l not in l2:
+      return False
+  return len(l1)==len(l2)
+
+class LAG(_Module):
+
+  """
+  link_pre_transform: Create virtual LAG links
+  """
+  def link_pre_transform(self, link: Box, topology: Box) -> None:
+    if log.debug_active('vlan'):
+      print(f'LAG link_pre_transform for {link}')
+    if 'lag' in link and link.get('type',"")!="lag":
+
+      # Lookup virtual LAG link with same id between same pair of nodes, create if not existing
+      vlag = None
+      for l in topology.links:
+        if 'lag' in l and l.get('type',"")=="lag" and l.lag.id == link.lag.id \
+            and same_list(l.interfaces,link.interfaces):
+          vlag = l
+          break
+      
+      if vlag is None:
+        vlag = data.get_box(link)
+        vlag.type = "lag"
+        vlag.linkindex = len(topology.links) + 1
+        vlag._linkname = f"links[{vlag.linkindex}]"
+        vlag.interfaces = [ i for i in link.interfaces ] # Make a deep copy, could use a set?
+        vlag.pop('mtu',None)                             # Remove any MTU attribute
```

Comment:
I'm pretty sure this will break multi-vendor OSPF

Reply:
I'll keep the mtu attribute, and skip setting it on bond interfaces in the platform specific scripts